integration-tests/twitter/src/main/java/org/apache/camel/quarkus/component/twitter/CamelRoute.java

```diff
@@ -16,25 +16,118 @@
  */
 package org.apache.camel.quarkus.component.twitter;
 
-import io.quarkus.runtime.annotations.RegisterForReflection;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.camel.Exchange;
 import org.apache.camel.builder.RouteBuilder;
-import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.apache.camel.component.twitter.TwitterConstants;
 
-@RegisterForReflection
 public class CamelRoute extends RouteBuilder {
 
-    @ConfigProperty(name = "twitter.user.name")
-    String twitterUserName;
+    private static final Set<String> TWITTER_SEARCH_HEADERS = new HashSet<>(
+            Arrays.asList(TwitterConstants.TWITTER_COUNT, TwitterConstants.TWITTER_KEYWORDS,
+                    TwitterConstants.TWITTER_MAXID, TwitterConstants.TWITTER_NUMBER_OF_PAGES,
+                    TwitterConstants.TWITTER_SEARCH_LANGUAGE, TwitterConstants.TWITTER_SINCEID));
+
+    private final StringBuilder ownTweets = new StringBuilder();
+    private final StringBuilder directMessages = new StringBuilder();
 
     @Override
     public void configure() {
-        from("twitter-timeline:user?user=ApacheCamel&count=1")
-            .to("log:timeline?showAll=true");
+        rest()
+            .get("/timeline")
+                /* Expose the polled messages */
+                .route()
+                .id("get-timeline")
+                .setBody(() -> {
+                    synchronized (ownTweets) {
+                        return ownTweets.toString();
+                    }
+                })
+                .endRest()
+            .post("/timeline")
+                .route()
+                .id("post-timeline")
+                .to("twitter-timeline://user")
+                .setHeader(Exchange.HTTP_RESPONSE_CODE, simple("201"))
+                .endRest()
+
+            .get("/search")
+                .route()
+                .id("get-search")
+                // The servlet component does not pass the query params as camel message headers for some reason
+                .process(e -> {
+                    final String query = e.getIn().getHeader("CamelHttpQuery", String.class);
+                    for (String kv : query.split("&")) {
+                        final String[] keyVal = kv.split("=");
+                        if (TWITTER_SEARCH_HEADERS.contains(keyVal[0])) {
+                            // Consider URL decoding if you copy this to production code
+                            e.getOut().setHeader(keyVal[0], keyVal[1]);
+                        }
+                    }
+                })
+                .to("twitter-search://_keywords_passed_via_CamelTwitterKeywords_header_")
+                .to("log:search?showAll=true")
+                .endRest()
+
+            .post("/directmessage")
+                .route()
+                .id("post-directmessage")
+                .to("twitter-directmessage://{{camel.component.twitter-directmessage.user}}")
```

Comment:
`camel.component.twitter-directmessage` is a syntax used to automatically bind properties to a component so I would use a property like `twitter.user` to avoid confusion

Reply:
Tried the original `twitter.user.name` and it did not work. IIRC the PropertyBindingSupport was complaining that the prop is not available. The JavaDoc of `PropertyBindingSupport.bindProperties()` says "Binds the properties with the given prefix to the target object" so it seems to consciously ignore all other props.

Reply:
properties binding support should not be involved here, it is just properties resolution in an endpoint. you may need to prefix the property with ‘integration.’ as at the moment there’s no full support for mo config (I’m working on it). using camel as prefix works as it is one of the whitelist prefix

Reply:
properties binding support should not be involved here, it is just properties resolution in an endpoint. you may need to prefix the property with ‘integration.’ as at the moment there’s no full support for mo config (I’m working on it). using camel as prefix works as it is one of the whitelist prefix